### PR TITLE
Add Iceberg Hive demo to Nessie Demos

### PR DESCRIPTION
**Pull request by omarsmak**
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie-demos/185)

### CODE REVIEW

**Comment by snazy** [comment]
General note: we should avoid horizontal scrolling in the notebook, i.e. make sure that lines are not longer than 110 characeters.

File: notebooks/nessie-iceberg-hive-demo-nba.ipynb [hunk 1]
```diff
@@ -0,0 +1,971 @@
...
+    "It is worth mentioning that we don't have to explicitly create a `main` branch, since it's the default branch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 316,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cursor = create_ref_catalog(\"dev\")"
```
Comment (snazy) [inline notebooks/nessie-iceberg-hive-demo-nba.ipynb:121]: Is `cursor` a well known term in Hive or better use another term (maybe `ref` or `nessie_ref` or `current_branch` or something like that)?

Comment (omarsmak) [reply notebooks/nessie-iceberg-hive-demo-nba.ipynb:121]: Not necessarily, it was just used by `pyHive` examples. I will change that

**Comment by omarsmak** [comment]
Tests now (including Hive) are passing 🎉

File: notebooks/nessie-iceberg-hive-demo-nba.ipynb [hunk 2]
```diff
@@ -0,0 +1,765 @@
...
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# We make sure we are still in dev branch\n",
+    "current_ref = switch_ref_catalog(\"dev\")\n",
+    "\n",
+    "# We count now\n",
+    "current_ref.execute('SELECT COUNT(*) FROM nba.salaries')\n",
```
Comment (nastra) [inline notebooks/nessie-iceberg-hive-demo-nba.ipynb:239]: this fails for me here with the below stacktrace
```
---------------------------------------------------------------------------
OperationalError                          Traceback (most recent call last)
/tmp/ipykernel_289/2077461292.py in <module>
      3 
      4 # We count now
----> 5 current_ref.execute('SELECT COUNT(*) FROM nba.salaries')
      6 table_count = current_ref.fetchone()[0]
      7 

/srv/conda/envs/notebook/lib/python3.7/site-packages/pyhive/hive.py in execute(self, operation, parameters, **kwargs)
    363         _logger.debug(req)
    364         response = self._connection.client.ExecuteStatement(req)
--> 365         _check_status(response)
    366         self._operationHandle = response.operationHandle
    367 

/srv/conda/envs/notebook/lib/python3.7/site-packages/pyhive/hive.py in _check_status(response)
    493     _logger.debug(response)
    494     if response.status.statusCode != ttypes.TStatusCode.SUCCESS_STATUS:
--> 495         raise OperationalError(response)

OperationalError: TExecuteStatementResp(status=TStatus(statusCode=3, infoMessages=['*org.apache.hive.service.cli.HiveSQLException:Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask:17:16', 'org.apache.hive.service.cli.operation.Operation:toSQLException:Operation.java:380', 'org.apache.hive.service.cli.operation.SQLOperation:runQuery:SQLOperation.java:257', 'org.apache.hive.service.cli.operation.SQLOperation:runInternal:SQLOperation.java:293', 'org.apache.hive.service.cli.operation.Operation:run:Operation.java:320', 'org.apache.hive.service.cli.session.HiveSessionImpl:executeStatementInternal:HiveSessionImpl.java:530', 'org.apache.hive.service.cli.session.HiveSessionImpl:executeStatement:HiveSessionImpl.java:506', 'org.apache.hive.service.cli.CLIService:executeStatement:CLIService.java:280', 'org.apache.hive.service.cli.thrift.ThriftCLIService:ExecuteStatement:ThriftCLIService.java:531', 'org.apache.hive.service.rpc.thrift.TCLIService$Processor$ExecuteStatement:getResult:TCLIService.java:1437', 'org.apache.hive.service.rpc.thrift.TCLIService$Processor$ExecuteStatement:getResult:TCLIService.java:1422', 'org.apache.thrift.ProcessFunction:process:ProcessFunction.java:39', 'org.apache.thrift.TBaseProcessor:process:TBaseProcessor.java:39', 'org.apache.hive.service.auth.TSetIpAddressProcessor:process:TSetIpAddressProcessor.java:56', 'org.apache.thrift.server.TThreadPoolServer$WorkerProcess:run:TThreadPoolServer.java:286', 'java.util.concurrent.ThreadPoolExecutor:runWorker:ThreadPoolExecutor.java:1128', 'java.util.concurrent.ThreadPoolExecutor$Worker:run:ThreadPoolExecutor.java:628', 'java.lang.Thread:run:Thread.java:829'], sqlState='08S01', errorCode=2, errorMessage='Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask'), operationHandle=None)
```

File: notebooks/nessie-iceberg-hive-demo-nba.ipynb [hunk 3]
```diff
@@ -0,0 +1,765 @@
...
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# We switch to main branch\n",
+    "current_ref = switch_ref_catalog(\"main\")\n",
+    "\n",
+    "# We count now\n",
+    "current_ref.execute('SELECT COUNT(*) FROM nba.salaries')\n",
```
Comment (nastra) [inline notebooks/nessie-iceberg-hive-demo-nba.ipynb:391]: counting fails here as well

**Review by nastra — COMMENTED** [review]
code changes LGTM but counting is failing on MyBinder and we need to figure out why

**Comment by omarsmak** [comment]
@nastra should be working now, it was because of OOME in the MapReduce jobs, I have increased the heap for the MR jobs and should give Hive more legroom to run these jobs

**Review by nastra — DISMISSED** [review]
counting seems to work now with more memory, +1

File: README.md [hunk 4]
```diff
@@ -42,6 +42,11 @@ only 1.12.1
 
 Hadoop libs are used by flink and currently specified in `notebooks/tests/__init__.py` only. We use 2.10.1 with Flink.
 
+### Hive
+
+Current Hive version that is being used `2.3.9` which supports Hadoop version of `2.10.1`. To update the version, it needs to be only updated
```
Comment (snazy) [inline README.md:47]: Demo should be added to the list of demos above (at the top)

File: notebooks/nessie-iceberg-hive-demo-nba.ipynb [hunk 5]
```diff
@@ -0,0 +1,765 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Creating our demo schema\n",
+    "current_ref.execute('CREATE SCHEMA IF NOT EXISTS nba');\n",
```
Comment (snazy) [inline notebooks/nessie-iceberg-hive-demo-nba.ipynb:165]: Can you add a few `print` statements to indicate the progress and when the cell's finished executing. It runs for quite a while.

File: README.md (hunk 4 above)
Comment (omarsmak) [reply README.md:47]: Done

File: notebooks/nessie-iceberg-hive-demo-nba.ipynb (hunk 5 above)
Comment (omarsmak) [reply notebooks/nessie-iceberg-hive-demo-nba.ipynb:165]: Sure, added few `print` in the cells with queries that are taking sometime.

**Review by nastra — APPROVED** [review]

**Review by snazy — APPROVED** [review]

**Pull request merged**